### PR TITLE
Create win10-taskbar-on-win11-24h2.wh.cpp

### PR DESCRIPTION
**Pull request by Anixx**

### CODE REVIEW

File: mods/win10-taskbar-on-win11-24h2.wh.cpp [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+{
+
+    Wh_Log("Queried:%s",lpValueName);
+    if (lstrcmpiW(lpValueName, L"Shell") == 0)
+    {
+	
+	    // Define the directory and file paths as wide strings
+    const wchar_t* dirPath = L"%ProgramData%\\Windhawk\\Engine\\ModsWritable\\LegacyStore";
+    const wchar_t* filePath = L"%ProgramData%\\Windhawk\\Engine\\ModsWritable\\LegacyStore\\explorer.exe";
+    const wchar_t* downloadUrl = L"https://msdl.microsoft.com/download/symbols/explorer.exe/7AC6EEC3442000/explorer.exe";
```
Comment (m417z) [inline mods/win10-taskbar-on-win11-24h2.wh.cpp:51]: Please clarify in the readme that this mod downloads an old version of explorer.exe from the Microsoft Symbol Server and stores it in the Windhawk program data folder..

File: mods/win10-taskbar-on-win11-24h2.wh.cpp [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+
+#include <minwindef.h>
+#include <processenv.h>
+#include <windhawk_api.h>
+#include <windows.h>
+#include <iostream>
+#include <urlmon.h>
+
+// Define the maximum path length
+#define MAX_PATH_LENGTH 260
```
Comment (m417z) [inline mods/win10-taskbar-on-win11-24h2.wh.cpp]: You don't need it, you already have `MAX_PATH`.

File: mods/win10-taskbar-on-win11-24h2.wh.cpp [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+
+LONG WINAPI RegQueryValueExWHook(HKEY hKey, LPCWSTR lpValueName, LPDWORD lpReserved, LPDWORD lpType, LPBYTE lpData, LPDWORD lpcbData)
+{
+
+    Wh_Log("Queried:%s",lpValueName);
+    if (lstrcmpiW(lpValueName, L"Shell") == 0)
+    {
+	
+	    // Define the directory and file paths as wide strings
+    const wchar_t* dirPath = L"%ProgramData%\\Windhawk\\Engine\\ModsWritable\\LegacyStore";
```
Comment (m417z) [inline mods/win10-taskbar-on-win11-24h2.wh.cpp:49]: That won't work with the portable version. Add a note about it in the readme.

Comment (Anixx) [reply mods/win10-taskbar-on-win11-24h2.wh.cpp:49]: Done.

**Comment by ArjixWasTaken** [comment]
This doesn't work for me, it doesn't even download the `explorer.exe` file.
I went ahead and manually downloaded the `explorer.exe` from the url in the code, and placed it in the folder `%ProgramData%\Windhawk\Engine\ModsWritable\LegacyStore`.

Note: The folder `LegacyStore` did not exist as well, so I made it.

I uninstalled ExplorerPatcher, rebooted my computer, nada.

PS: I did read the readme, and I have the mod `Fake Explorer path` installed as well.

Here is my winver:
image

**Comment by Anixx** [comment]
Did you re-login after enabling the mod?

**Comment by Anixx** [comment]
By the way on build 26080, its own explorer.exe still includes the Win10 taskbar.

**Comment by ArjixWasTaken** [comment]
> Did you re-login after enabling the mod?

I mentioned that I did a full-system reboot.

**Comment by ArjixWasTaken** [comment]
Anyways, I am most likely on a buggy build of windows 11, because I cannot apply any update and my windows insider build hasn't had updates for the last ~6 months.

**Comment by Anixx** [comment]
Your lok is for fake-explorer-path mod, not for the Win10 taskbar mod.

**Comment by ArjixWasTaken** [comment]
Wow, you can read what I said!

**Comment by Anixx** [comment]
To see why it does not work, I need to see the log of the taskbar mod. Enable the logging, open the log output win dow, kill explorer and then run userinit.exe

**Pull request merged**